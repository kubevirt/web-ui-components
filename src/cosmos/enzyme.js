import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
