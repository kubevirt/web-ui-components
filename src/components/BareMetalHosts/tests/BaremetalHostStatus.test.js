import React from 'react';
import { shallow } from 'enzyme';

import BaremetalHostStatusFixture from '../fixtures/BaremetalHostStatus.fixture';

import { BaremetalHostStatus } from '..';

const getComponentFunction = props => () => <BaremetalHostStatus {...props} />;

describe('<BaremetalHostStatus />', () => {
  BaremetalHostStatusFixture.forEach((fixture, index) => {
    it(`renders the correct subcomponent for the "${fixture.props.host.status.provisioning.state}" state (${index})`, () => {
      const component = shallow(getComponentFunction(fixture.props)());
      expect(component).toMatchSnapshot();
    });
  });
});
