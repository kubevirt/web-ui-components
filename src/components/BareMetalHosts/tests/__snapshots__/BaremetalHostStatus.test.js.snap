// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaremetalHostStatus /> renders the correct subcomponent for the "discovered" state (5) 1`] = `
<AddDiscoveredHostLink
  host={
    Object {
      "status": Object {
        "provisioning": Object {
          "state": "discovered",
        },
      },
    }
  }
/>
`;

exports[`<BaremetalHostStatus /> renders the correct subcomponent for the "match profile" state (2) 1`] = `
<GenericProgress
  status="match profile"
  text="Matching profile"
/>
`;

exports[`<BaremetalHostStatus /> renders the correct subcomponent for the "power management error" state (4) 1`] = `
<GenericError
  status="power management error"
  text="Power Management Error"
/>
`;

exports[`<BaremetalHostStatus /> renders the correct subcomponent for the "provisioned" state (1) 1`] = `
<GenericSuccess
  status="provisioned"
  text="Provisioned"
/>
`;

exports[`<BaremetalHostStatus /> renders the correct subcomponent for the "provisioning error" state (3) 1`] = `
<GenericError
  status="provisioning error"
  text="Provisioning Error"
/>
`;

exports[`<BaremetalHostStatus /> renders the correct subcomponent for the "unknown state" state (0) 1`] = `
<GenericStatus
  status="unknown state"
  text="unknown state"
/>
`;
